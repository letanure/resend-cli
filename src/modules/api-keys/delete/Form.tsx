import { Spinner } from '@inkjs/ui';
import { Box, useInput } from 'ink';
import React, { useState } from 'react';
import { SimpleForm } from '@/components/forms/SimpleForm.js';
import { ErrorScreen } from '@/components/ui/ErrorScreen.js';
import { Layout } from '@/components/ui/layout.js';
import { SuccessScreen } from '@/components/ui/SuccessScreen.js';
import { config } from '@/config/config.js';
import { useDryRun } from '@/contexts/DryRunProvider.js';
import { useResend } from '@/contexts/ResendProvider.js';
import { useApiKeySelector } from '@/hooks/useApiKeySelector.js';
import { deleteApiKey } from './action.js';
import { fields } from './fields.js';
import { type DeleteApiKeyData, deleteApiKeySchema } from './schema.js';

interface FormProps {
	onExit: () => void;
}

export const Form = ({ onExit }: FormProps) => {
	const { isDryRun } = useDryRun();
	const { apiKey } = useResend();
	const [isSubmitting, setIsSubmitting] = useState(false);
	const [successData, setSuccessData] = useState<Record<string, unknown> | null>(null);
	const [isDryRunSuccess, setIsDryRunSuccess] = useState(false);
	const [error, setError] = useState<{ title: string; message: string; suggestion?: string } | null>(null);
	const [selectedApiKeyId, setSelectedApiKeyId] = useState<string>('');

	const initialFormData = React.useMemo(() => {
		return selectedApiKeyId ? { api_key_id: selectedApiKeyId } : undefined;
	}, [selectedApiKeyId]);

	const apiKeySelector = useApiKeySelector((apiKeyId) => {
		setSelectedApiKeyId(apiKeyId);
	});

	// Create form fields with API key selector
	const formFields = React.useMemo(() => {
		return fields.map((field) => {
			if (field.name === 'api_key_id') {
				return {
					...field,
					type: 'input-with-selector' as const,
					onSelectorOpen: apiKeySelector.openSelector,
				};
			}
			return field;
		});
	}, [apiKeySelector.openSelector]);

	// Handle Esc/Left arrow key to go back from result screens
	useInput(
		(_input, key) => {
			if ((key.escape || key.leftArrow) && (successData || error)) {
				setSuccessData(null);
				setIsDryRunSuccess(false);
				setError(null);
			}
		},
		{ isActive: !!(successData || error) },
	);

	const handleSubmit = async (data: DeleteApiKeyData) => {
		setIsSubmitting(true);
		try {
			if (isDryRun) {
				setSuccessData({
					'API Key ID': data.api_key_id,
					'API Key': apiKey ? `${apiKey.substring(0, 10)}...` : 'Not set',
					Status: 'Validation successful! (API key not deleted due to dry-run mode)',
				});
				setIsDryRunSuccess(true);
			} else {
				const result = await deleteApiKey(data, apiKey);

				if (result.success && result.data) {
					setSuccessData({
						'API Key ID': data.api_key_id,
						'Object Type': 'api_key',
						Status: 'Deleted successfully',
					});
					setIsDryRunSuccess(false);
				} else {
					setError({
						title: 'API Key Deletion Failed',
						message: result.error || 'Unknown error occurred',
						suggestion: 'Check the API key ID and ensure it exists in your Resend account',
					});
				}
			}
		} catch (error) {
			setError({
				title: 'API Key Deletion Error',
				message: error instanceof Error ? error.message : 'Unknown error',
				suggestion: 'Please check your API key and network connection',
			});
		} finally {
			setIsSubmitting(false);
		}
	};

	if (isSubmitting) {
		return (
			<Layout
				headerText={`${config.baseTitle} - API Keys - Delete`}
				showNavigationInstructions={false}
				navigationContext="none"
			>
				<Spinner label="Deleting API key..." />
			</Layout>
		);
	}

	if (successData) {
		return (
			<SuccessScreen
				data={successData}
				successMessage="API Key Deleted Successfully"
				headerText={`${config.baseTitle} - API Keys - Delete`}
				isDryRun={isDryRunSuccess}
				onExit={() => {
					setSuccessData(null);
					setIsDryRunSuccess(false);
					onExit();
				}}
			/>
		);
	}

	if (error) {
		return (
			<ErrorScreen
				title={error.title}
				message={error.message}
				suggestion={error.suggestion}
				headerText={`${config.baseTitle} - API Keys - Delete`}
				onExit={() => {
					setError(null);
					onExit();
				}}
				showRetry={true}
				onRetry={() => {
					setError(null);
					setIsSubmitting(false);
				}}
			/>
		);
	}

	// If the API key selector is open, render it instead of the form
	if (apiKeySelector.isOpen) {
		return apiKeySelector.selectorComponent;
	}

	return (
		<Layout
			headerText={`${config.baseTitle} - API Keys - Delete`}
			showNavigationInstructions={true}
			navigationContext="form-single"
		>
			<Box flexDirection="column">
				<SimpleForm<DeleteApiKeyData>
					fields={formFields}
					onSubmit={handleSubmit}
					onCancel={onExit}
					validateWith={deleteApiKeySchema}
					initialData={initialFormData}
				/>
			</Box>
		</Layout>
	);
};
