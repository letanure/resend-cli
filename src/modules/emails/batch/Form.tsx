import { Alert } from '@inkjs/ui';
import { Box, Text, useInput } from 'ink';
import { Layout } from '@/components/ui/layout.js';
import { config } from '@/config/config.js';

interface FormProps {
	onExit: () => void;
}

export const Form = ({ onExit }: FormProps) => {
	useInput((input, key) => {
		if (input === 'q' || key.escape) {
			onExit();
		}
	});

	return (
		<Layout headerText={`${config.baseTitle} - Emails - Batch Send`}>
			<Box flexDirection="column" gap={1}>
				<Alert variant="info">Batch email functionality is coming soon!</Alert>
				<Text>This feature will allow you to send multiple emails in a single request.</Text>
				<Box marginTop={1}>
					<Text>
						Press <Text color="yellow">Esc</Text> or <Text color="yellow">q</Text> to go back
					</Text>
				</Box>
			</Box>
		</Layout>
	);
};
