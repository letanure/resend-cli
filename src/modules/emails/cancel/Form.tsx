import { Spinner } from '@inkjs/ui';
import { Box, Text, useInput } from 'ink';
import { useState } from 'react';
import { SimpleForm } from '@/components/forms/SimpleForm.js';
import { ErrorDisplay } from '@/components/ui/ErrorDisplay.js';
import { Layout } from '@/components/ui/layout.js';
import { config } from '@/config/config.js';
import { useDryRun } from '@/contexts/DryRunProvider.js';
import { useResend } from '@/contexts/ResendProvider.js';
import { EmailDisplay } from '../shared/EmailDisplay.js';
import { cancelEmail } from './action.js';
import { fields } from './fields.js';
import { CancelEmailOptionsSchema, type CancelEmailOptionsType } from './schema.js';

interface EmailCancelFormProps {
	onExit: () => void;
}

export const EmailCancelForm = ({ onExit }: EmailCancelFormProps) => {
	const { isDryRun } = useDryRun();
	const { apiKey } = useResend();
	const [isSubmitting, setIsSubmitting] = useState(false);
	const [cancelResult, setCancelResult] = useState<Record<string, unknown> | null>(null);
	const [showDryRunData, setShowDryRunData] = useState<Record<string, unknown> | null>(null);
	const [error, setError] = useState<{ title: string; message: string; suggestion?: string } | null>(null);

	// Handle Esc key to go back from result screens
	useInput(
		(_input, key) => {
			if (key.escape && (cancelResult || showDryRunData || error)) {
				setCancelResult(null);
				setShowDryRunData(null);
				setError(null);
			}
		},
		{ isActive: !!(cancelResult || showDryRunData || error) },
	);

	const handleSubmit = async (data: CancelEmailOptionsType) => {
		setIsSubmitting(true);
		try {
			if (isDryRun) {
				setShowDryRunData({
					'Email ID': data.id,
					'API Key': apiKey ? `${apiKey.substring(0, 10)}...` : 'Not set',
					'Dry Run': 'true',
					Status: 'Validation successful! (Email not cancelled due to dry-run mode)',
				});
			} else {
				const result = await cancelEmail(data.id, apiKey);

				if (result.success && result.data) {
					setCancelResult({
						'Email ID': data.id,
						'Cancel Status': 'Successfully cancelled',
						'Object Type': result.data.object,
						'Cancelled At': new Date().toISOString(),
					});
				} else {
					setError({
						title: 'Email Cancellation Failed',
						message: result.error || 'Unknown error occurred',
						suggestion: 'Check the email ID and ensure it exists and is scheduled for future delivery',
					});
				}
			}
		} catch (error) {
			setError({
				title: 'Email Cancellation Error',
				message: error instanceof Error ? error.message : 'Unknown error',
				suggestion: 'Please check your API key and network connection',
			});
		} finally {
			setIsSubmitting(false);
		}
	};

	if (isSubmitting) {
		return (
			<Layout headerText={`${config.baseTitle} - Emails - Cancel`}>
				<Spinner label="Cancelling email..." />
			</Layout>
		);
	}

	if (cancelResult) {
		return (
			<Layout headerText={`${config.baseTitle} - Emails - Cancel - Success`}>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={cancelResult} title="Email Cancelled Successfully" />
					</Box>
					<Box>
						<Text dimColor={true}>Press Esc to go back</Text>
					</Box>
				</Box>
			</Layout>
		);
	}

	if (showDryRunData) {
		return (
			<Layout headerText={`${config.baseTitle} - Emails - Cancel - Dry Run`}>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={showDryRunData} title="DRY RUN - Email cancellation data (validation only)" />
					</Box>
					<Box>
						<Text dimColor={true}>Press Esc to go back</Text>
					</Box>
				</Box>
			</Layout>
		);
	}

	if (error) {
		return (
			<Layout headerText={`${config.baseTitle} - Emails - Cancel - Error`}>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<ErrorDisplay title={error.title} message={error.message} suggestion={error.suggestion} />
					</Box>
					<Box>
						<Text dimColor={true}>Press Esc to go back</Text>
					</Box>
				</Box>
			</Layout>
		);
	}

	return (
		<Layout headerText={`${config.baseTitle} - Emails - Cancel`}>
			<SimpleForm<CancelEmailOptionsType>
				fields={fields}
				onSubmit={handleSubmit}
				onCancel={onExit}
				validateWith={CancelEmailOptionsSchema}
			/>
		</Layout>
	);
};
