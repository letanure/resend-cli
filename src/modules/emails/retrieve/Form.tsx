import { Spinner } from '@inkjs/ui';
import { Box, useInput } from 'ink';
import { useState } from 'react';
import { SimpleForm } from '@/components/forms/SimpleForm.js';
import { ErrorScreen } from '@/components/ui/ErrorScreen.js';
import { Layout } from '@/components/ui/layout.js';
import { config } from '@/config/config.js';
import { useDryRun } from '@/contexts/DryRunProvider.js';
import { useResend } from '@/contexts/ResendProvider.js';
import { EmailDisplay } from '../shared/EmailDisplay.js';
import { EMAIL_FULL_FIELDS } from '../shared/fields.js';
import { getEmail } from './action.js';
import { fields } from './fields.js';
import { GetEmailOptionsSchema, type GetEmailOptionsType } from './schema.js';

interface FormProps {
	onExit: () => void;
}

export const Form = ({ onExit }: FormProps) => {
	const { isDryRun } = useDryRun();
	const { apiKey } = useResend();
	const [isSubmitting, setIsSubmitting] = useState(false);
	const [emailData, setEmailData] = useState<Record<string, unknown> | null>(null);
	const [showDryRunData, setShowDryRunData] = useState<Record<string, unknown> | null>(null);
	const [error, setError] = useState<{ title: string; message: string; suggestion?: string } | null>(null);

	// Handle Esc/Left arrow key to go back from result screens
	useInput(
		(_input, key) => {
			if ((key.escape || key.leftArrow) && (emailData || showDryRunData || error)) {
				setEmailData(null);
				setShowDryRunData(null);
				setError(null);
			}
		},
		{ isActive: !!(emailData || showDryRunData || error) },
	);

	const handleSubmit = async (data: GetEmailOptionsType) => {
		setIsSubmitting(true);
		try {
			if (isDryRun) {
				setShowDryRunData({
					'Email ID': data.id,
					'API Key': apiKey ? `${apiKey.substring(0, 10)}...` : 'Not set',
					'Dry Run': 'true',
					Status: 'Validation successful! (Email not retrieved due to dry-run mode)',
				});
			} else {
				const result = await getEmail(data.id, apiKey);

				if (result.success && result.data) {
					setEmailData(result.data as unknown as Record<string, unknown>);
				} else {
					setError({
						title: 'Email Retrieval Failed',
						message: result.error || 'Unknown error occurred',
						suggestion: 'Check the email ID and ensure it exists in your Resend account',
					});
				}
			}
		} catch (error) {
			setError({
				title: 'Email Retrieval Error',
				message: error instanceof Error ? error.message : 'Unknown error',
				suggestion: 'Please check your API key and network connection',
			});
		} finally {
			setIsSubmitting(false);
		}
	};

	if (isSubmitting) {
		return (
			<Layout headerText={`${config.baseTitle} - Emails - Retrieve`}>
				<Spinner label="Retrieving email..." />
			</Layout>
		);
	}

	if (emailData) {
		return (
			<Layout
				headerText={`${config.baseTitle} - Emails - Retrieve - Success`}
				showNavigationInstructions={true}
				navigationContext="result"
			>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={emailData} title="Email Retrieved Successfully" fieldsToShow={EMAIL_FULL_FIELDS} />
					</Box>
				</Box>
			</Layout>
		);
	}

	if (showDryRunData) {
		return (
			<Layout
				headerText={`${config.baseTitle} - Emails - Retrieve - Dry Run`}
				showNavigationInstructions={true}
				navigationContext="result"
			>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={showDryRunData} title="DRY RUN - Email retrieval data (validation only)" />
					</Box>
				</Box>
			</Layout>
		);
	}

	if (error) {
		return (
			<ErrorScreen
				title={error.title}
				message={error.message}
				suggestion={error.suggestion}
				headerText={`${config.baseTitle} - Emails - Retrieve`}
				onExit={() => {
					setError(null);
					onExit();
				}}
				showRetry={true}
				onRetry={() => {
					setError(null);
					setIsSubmitting(false);
				}}
			/>
		);
	}

	return (
		<Layout
			headerText={`${config.baseTitle} - Emails - Retrieve`}
			showNavigationInstructions={true}
			navigationContext="form-single"
		>
			<SimpleForm<GetEmailOptionsType>
				fields={fields}
				onSubmit={handleSubmit}
				onCancel={onExit}
				validateWith={GetEmailOptionsSchema}
			/>
		</Layout>
	);
};
