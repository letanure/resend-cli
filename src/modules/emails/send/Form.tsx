import type { CreateEmailOptions } from 'resend';
import { SimpleForm } from '@/components/forms/SimpleForm.js';
import { Layout } from '@/components/ui/layout.js';
import { config } from '@/config/config.js';
import { useDryRun } from '@/contexts/DryRunProvider.js';
import { useResend } from '@/contexts/ResendProvider.js';
import { sendEmail } from './action.js';
import { fields } from './fields.js';
import { CreateEmailOptionsSchema, type CreateEmailOptionsType } from './schema.js';

interface FormProps {
	onExit: () => void;
	onEmailSent?: (emailId: string) => void;
	onEmailError?: (error: string) => void;
}

export const Form = ({ onExit, onEmailSent, onEmailError }: FormProps) => {
	const { apiKey } = useResend();
	const { isDryRun } = useDryRun();

	const handleSubmit = async (validatedData: CreateEmailOptionsType) => {
		// In dry-run mode, just log the data and exit
		if (isDryRun) {
			console.log('🔍 DRY RUN - Email data validated:');
			console.log(JSON.stringify(validatedData, null, 2));
			console.log('✅ Validation successful! (Email not sent due to dry-run mode)');
			onExit();
			return;
		}

		// Data is already validated and transformed by SimpleForm + Zod schema
		// Type assertion is safe here because Zod validation ensures compatibility
		const result = await sendEmail(validatedData as CreateEmailOptions, apiKey);

		if (result.success && result.data?.id) {
			if (onEmailSent) {
				onEmailSent(result.data.id);
			} else {
				console.log(`✅ Email sent successfully! ID: ${result.data.id}`);
				onExit();
			}
			return;
		}

		if (onEmailError) {
			const errorMessage = result.error || 'Unknown error occurred';
			onEmailError(errorMessage);
		} else {
			console.error(`❌ Failed to send email: ${result.error}`);
		}
	};

	return (
		<Layout headerText={`${config.baseTitle} - Emails - Send`}>
			<SimpleForm<CreateEmailOptionsType>
				fields={fields}
				validateWith={CreateEmailOptionsSchema}
				onSubmit={handleSubmit}
				onCancel={onExit}
			/>
		</Layout>
	);
};
