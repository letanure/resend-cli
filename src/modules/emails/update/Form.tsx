import { Spinner } from '@inkjs/ui';
import { Box, Text, useInput } from 'ink';
import { useState } from 'react';
import { SimpleForm } from '@/components/forms/SimpleForm.js';
import { ErrorScreen } from '@/components/ui/ErrorScreen.js';
import { Layout } from '@/components/ui/layout.js';
import { config } from '@/config/config.js';
import { useDryRun } from '@/contexts/DryRunProvider.js';
import { useResend } from '@/contexts/ResendProvider.js';
import { EmailDisplay } from '../shared/EmailDisplay.js';
import { updateEmail } from './action.js';
import { fields } from './fields.js';
import { UpdateEmailOptionsSchema, type UpdateEmailOptionsType } from './schema.js';

interface FormProps {
	onExit: () => void;
}

export const Form = ({ onExit }: FormProps) => {
	const { isDryRun } = useDryRun();
	const { apiKey } = useResend();
	const [isSubmitting, setIsSubmitting] = useState(false);
	const [updateResult, setUpdateResult] = useState<Record<string, unknown> | null>(null);
	const [showDryRunData, setShowDryRunData] = useState<Record<string, unknown> | null>(null);
	const [error, setError] = useState<{ title: string; message: string; suggestion?: string } | null>(null);

	// Handle Esc key to go back from result screens
	useInput(
		(_input, key) => {
			if (key.escape && (updateResult || showDryRunData || error)) {
				setUpdateResult(null);
				setShowDryRunData(null);
				setError(null);
			}
		},
		{ isActive: !!(updateResult || showDryRunData || error) },
	);

	const handleSubmit = async (data: UpdateEmailOptionsType) => {
		setIsSubmitting(true);
		try {
			if (isDryRun) {
				setShowDryRunData({
					'Email ID': data.id,
					'New Scheduled Date': data.scheduledAt,
					'API Key': apiKey ? `${apiKey.substring(0, 10)}...` : 'Not set',
					'Dry Run': 'true',
					Status: 'Validation successful! (Email not updated due to dry-run mode)',
				});
			} else {
				const result = await updateEmail(data, apiKey);

				if (result.success && result.data) {
					setUpdateResult({
						'Email ID': data.id,
						'New Scheduled Date': data.scheduledAt,
						'Update Status': 'Successfully updated',
						'Object Type': result.data.object,
						'Updated At': new Date().toISOString(),
					});
				} else {
					setError({
						title: 'Email Update Failed',
						message: result.error || 'Unknown error occurred',
						suggestion: 'Check the email ID and ensure it exists and is scheduled for future delivery',
					});
				}
			}
		} catch (error) {
			setError({
				title: 'Email Update Error',
				message: error instanceof Error ? error.message : 'Unknown error',
				suggestion: 'Please check your API key and network connection',
			});
		} finally {
			setIsSubmitting(false);
		}
	};

	if (isSubmitting) {
		return (
			<Layout
				headerText={`${config.baseTitle} - Emails - Update`}
				showNavigationInstructions={false}
				navigationContext="none"
			>
				<Spinner label="Updating email..." />
			</Layout>
		);
	}

	if (updateResult) {
		return (
			<Layout
				headerText={`${config.baseTitle} - Emails - Update - Success`}
				showNavigationInstructions={true}
				navigationContext="result"
			>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={updateResult} title="Email Updated Successfully" />
					</Box>
					<Box>
						<Text dimColor={true}>Press Esc to go back</Text>
					</Box>
				</Box>
			</Layout>
		);
	}

	if (showDryRunData) {
		return (
			<Layout
				headerText={`${config.baseTitle} - Emails - Update - Dry Run`}
				showNavigationInstructions={true}
				navigationContext="result"
			>
				<Box flexDirection="column">
					<Box marginBottom={1}>
						<EmailDisplay data={showDryRunData} title="DRY RUN - Email update data (validation only)" />
					</Box>
					<Box>
						<Text dimColor={true}>Press Esc to go back</Text>
					</Box>
				</Box>
			</Layout>
		);
	}

	if (error) {
		return (
			<ErrorScreen
				title={error.title}
				message={error.message}
				suggestion={error.suggestion}
				headerText={`${config.baseTitle} - Emails - Update`}
				onExit={() => {
					setError(null);
					onExit();
				}}
				showRetry={true}
				onRetry={() => {
					setError(null);
					setIsSubmitting(false);
				}}
			/>
		);
	}

	return (
		<Layout
			headerText={`${config.baseTitle} - Emails - Update`}
			showNavigationInstructions={true}
			navigationContext="form"
		>
			<SimpleForm<UpdateEmailOptionsType>
				fields={fields}
				onSubmit={handleSubmit}
				onCancel={onExit}
				validateWith={UpdateEmailOptionsSchema}
			/>
		</Layout>
	);
};
