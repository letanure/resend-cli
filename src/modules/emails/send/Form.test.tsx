import { describe, expect, it, vi } from 'vitest';
import { renderWithProviders } from '@/utils/test-utils.js';
import { Form } from './Form.js';

describe('Email Send Form', () => {
	const mockOnExit = vi.fn();

	it('renders form with email fields', () => {
		const { lastFrame } = renderWithProviders(<Form onExit={mockOnExit} />);

		const output = lastFrame();
		expect(output).toContain('From');
		expect(output).toContain('To');
		expect(output).toContain('Subject');
		expect(output).toContain('HTML Content');
		expect(output).toContain('Plain Text');
	});

	it('exits when ESC is pressed', async () => {
		const { stdin } = renderWithProviders(<Form onExit={mockOnExit} />);

		await stdin.write('\x1B'); // ESC key

		expect(mockOnExit).toHaveBeenCalled();
	});

	it('shows dry-run banner in dry-run mode', () => {
		const { lastFrame } = renderWithProviders(<Form onExit={mockOnExit} />, { isDryRun: true });

		expect(lastFrame()).toContain('DRY RUN MODE');
	});

	it('shows validation errors when submitting empty form', async () => {
		const { lastFrame, stdin } = renderWithProviders(<Form onExit={mockOnExit} />);

		// Submit without filling fields
		await stdin.write('\r'); // Enter key
		await new Promise((resolve) => setTimeout(resolve, 50));

		const output = lastFrame();
		expect(output).toContain('Required'); // Validation error
	});

	it('renders correctly in dry-run mode', () => {
		const { lastFrame } = renderWithProviders(<Form onExit={mockOnExit} />, { isDryRun: true });

		const output = lastFrame();
		expect(output).toContain('DRY RUN MODE');
		expect(output).toContain('From');
	});

	it('renders correctly in normal mode', () => {
		const { lastFrame } = renderWithProviders(<Form onExit={mockOnExit} />, { isDryRun: false });

		const output = lastFrame();
		expect(output).not.toContain('DRY RUN MODE');
		expect(output).toContain('From');
	});
});
