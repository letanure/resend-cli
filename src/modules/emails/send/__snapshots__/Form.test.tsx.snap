// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Email Send Form > matches snapshot in dry-run mode 1`] = `
"
 ðŸ“§ Resend CLI - Emails - Send

 â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
 â”‚ âš   DRY RUN MODE: Operations will be validated but not executed.                                â”‚
 â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯


 From           â”‚our Name <sender@domain.com>
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
                Sender email address with optional name
 To             recipient@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Subject        Enter email subject
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 BCC            bcc@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 CC             cc@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Schedule       in 1 hour OR 2024-08-05T11:52:01.858Z
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Reply To       reply@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 HTML Content   <p>Your HTML content here</p>
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Plain Text     Your plain text message here...
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€


 Tab/â†“ Next field Â· Shift+Tab/â†‘ Previous field
 Enter Submit Â· Esc Cancel

"
`;

exports[`Email Send Form > matches snapshot in normal mode 1`] = `
"
 ðŸ“§ Resend CLI - Emails - Send

 From           â”‚our Name <sender@domain.com>
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
                Sender email address with optional name
 To             recipient@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Subject        Enter email subject
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 BCC            bcc@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 CC             cc@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Schedule       in 1 hour OR 2024-08-05T11:52:01.858Z
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Reply To       reply@domain.com
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 HTML Content   <p>Your HTML content here</p>
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€

 Plain Text     Your plain text message here...
                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€


 Tab/â†“ Next field Â· Shift+Tab/â†‘ Previous field
 Enter Submit Â· Esc Cancel

"
`;
